barman/retention_policies.py

```diff
@@ -185,6 +210,16 @@ def _backup_report(self, source):
                 i = i + 1
             else:
                 report[bid] = BackupInfo.NONE
+
+            if backups[bid].has_children:
+                status = report[bid]
+                if report[bid] in (BackupInfo.KEEP_STANDALONE, BackupInfo.KEEP_FULL):
```

Comment:
Here, if the parent has children but reported as KEEP_*, we have to propagate some status to the children, otherwise we will end up having that bug in list_buckets. list_buckets takes into account all available_backups. I thought that if parent has KEEP status, the incremental should be VALID since we only delete incrementals through the deletion of its parent. As soon as a parent gets obsolete, automatically the children will change status as well. So when a parent backup is VALID or KEEP_STANDALONE or KEEP_FULL, the children will be VALID. 
Please, what are your thoughts on this? Would this make sense?

Reply:
At a first glance I think it makes sense to have keep backups' children as `VALID`.

Reply:
Yeah VALID is the way to go imho.